src/Store/DoctrineDbalStore.php

```diff
@@ -423,4 +439,79 @@ private function executeSave(
             throw new UniqueConstraintViolation($e);
         }
     }
+
+    private function lock(): void
+    {
+        if (!$this->config['locking']) {
+            return;
+        }
+
+        $platform = $this->connection->getDatabasePlatform();
+
+        if ($platform instanceof PostgreSQLPlatform) {
+            $this->connection->executeStatement(
+                sprintf(
+                    'SELECT pg_advisory_lock(%s)',
+                    $this->config['lock_id'],
+                ),
+            );
+
+            return;
+        }
+
+        if ($platform instanceof MariaDBPlatform || $platform instanceof MySQLPlatform) {
+            $this->connection->fetchAllAssociative(
+                sprintf(
+                    'SELECT GET_LOCK("%s", %d)',
+                    $this->config['lock_id'],
+                    $this->config['lock_timeout'],
+                ),
+            );
+
+            return;
+        }
+
+        if ($platform instanceof SQLitePlatform) {
+            return; // locking is not supported
```

Comment:
I think more accurate would be that locking is here not needed, right? Otherwise where is the distinction between sql lite and other db platforms which do not have a locking implementation for our users?

Reply:
I have to look again at how it works with sqlite. I think it has a file lock by default, or you do it via file lock.

My idea was that DX is better, that you don't have to deal with locking when you switch between sqlite and other DBs for dev or testing. But I'm not sure about that.

Reply:
I think locking in SQLite is not needed due to the nature of how SQLite is working. What I meant here is that the comment is not quite correct. I would go a little more in detail that it is not needed and not `not supported`.